alacritty/src/ipc.rs

```diff
@@ -1,25 +1,30 @@
 //! Alacritty socket IPC.
 
 use std::ffi::OsStr;
-use std::io::{Error as IoError, ErrorKind, Result as IoResult};
-use std::os::unix::net::UnixDatagram;
+use std::io::{BufRead, BufReader, Error as IoError, ErrorKind, Result as IoResult, Write};
+use std::os::unix::net::{UnixListener, UnixStream};
 use std::path::PathBuf;
 use std::{env, fs, process};
 
 use glutin::event_loop::EventLoopProxy;
 use log::warn;
+use serde::{Deserialize, Serialize};
 
 use alacritty_terminal::thread;
 
-use crate::cli::Options;
+use crate::cli::{Options, TerminalOptions};
 use crate::event::{Event, EventType};
 
-/// IPC socket message for creating a new window.
-pub const SOCKET_MESSAGE_CREATE_WINDOW: [u8; 1] = [1];
-
 /// Environment variable name for the IPC socket path.
 const ALACRITTY_SOCKET_ENV: &str = "ALACRITTY_SOCKET";
 
+const JSON_SEPARATOR: u8 = b'\n';
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum IPCMessage {
+    CreateWindow(TerminalOptions),
```

Comment:
Do we want to somehow version the messages sent through the protocol? I think it should be fine without it since we just serde deserialize everything anyway and that will return an error if things don't match, but I'm curious if you've considered this.

Unfortunately the only interesting part for versioning would be forward-compatibility, since you'll rarely have a server that is newer than the client connecting to the socket. And that can't really be done anyway, so probably pointless.

Reply:
Not sure about this one.

Reply:
I personally think just not doing any versioning should be perfectly fine. Do you have any arguments for why it might be smart to do it?

Reply:
I'm fine without versioning for now, since feature isn't even released, could think about it close to release.